optimade/server/entry_collections.py

```diff
@@ -176,15 +175,16 @@ def _parse_params(
             )
 
         if getattr(params, "page_limit", False):
-            limit = self.page_limit
-            if params.page_limit != self.page_limit:
+            limit = CONFIG.page_limit
+            if params.page_limit != CONFIG.page_limit:
                 limit = params.page_limit
-            if limit > self.page_limit:
+            if limit > CONFIG.db_page_limit:
                 raise HTTPException(
-                    status_code=400, detail=f"Max page_limit is {self.page_limit}"
+                    status_code=403,  # Forbidden
+                    detail=f"Max allowed page_limit is {CONFIG.db_page_limit}, you requested {limit}",
                 )
             if limit == 0:
-                limit = self.page_limit
+                limit = CONFIG.page_limit
```

Comment:
is this necessary?
what if limit < 0?

If people explicitly request 0 entries, I'd say they should get 0 entries

Reply:
> what if limit < 0?

This is handled by the QueryParams, i.e., All query parameter of `page_*` are `NonNegativeInt` types.

Reply:
But I guess we could put in another check, although it shouldn't be necessary?